toolkit/cli/smart-contracts-commands/src/addresses.rs

```diff
@@ -0,0 +1,25 @@
+use partner_chains_cardano_offchain::{csl::NetworkTypeExt, scripts_data::get_scripts_data};
+use sidechain_domain::{NetworkType, UtxoId};
+
+use crate::NetworkArg;
+
+#[derive(Clone, Debug, clap::Parser)]
+pub struct AddressesCmd {
+	#[arg(long, short = 'n')]
+	network: NetworkArg,
```

Comment:
I think use ogmios connection data and read the network type from Ogmios.

Reply:
I thought about it, but I'm not sure. As it stands, the command can be used offline which is a bonus. Maybe I can change it so it requires either ogmios connection or network type so the user can choose